Cosmos DB: Added regions_contacted as new dimension and operation count metrics (#1493)
open-telemetry/semantic-conventions: docs/database/cosmosdb.md

| `db.operation.name`                  | `"read_item"` |
| `server.address`                     | `"account.documents.azure.com"` |
| `db.cosmosdb.client_id`              | `"3ba4827d-4422-483f-b59f-85b74211c11d"` |
| `user_agent.original`                | `"cosmos-netstandard-sdk/3.23.0\|3.23.1\|1\|X64\|Linux 5.4.0-1098-azure 104 18\|.NET Core 3.1.32\|S\|"` |
| `db.cosmosdb.request_content_length` | `20` |
| `db.response.status_code`            | `201` |
| `db.cosmosdb.sub_status_code`        | `0` |
| `db.cosmosdb.request_charge`         | `7.43` |

## Operation Level Metrics

The following metrics provide insights into Azure Cosmos DB client operation performance and behavior.

### Metric: `db.client.operation.count`

This metric is [required][MetricRequired].

> It captures the total number of operation made by an Azure Cosmos DB account. This metric follows the common [db.client.operation.count](/docs/database/database-metrics.md#metric-dbclientoperationduration) definition.

Reviewer: ```suggestion
It captures the total number of operations for an Azure Cosmos DB account. This metric follows the common [db.client.operation.count](/docs/database/database-metrics.md#metric-dbclientoperationduration) definition.
```

| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
| -------- | --------------- | ----------- | -------------- | --------- |
| `db.cosmosdb.operation.request_charge` | Histogram | `{request_unit}` | [Request charge](https://learn.microsoft.com/azure/cosmos-db/request-units) consumed by the operation | ![Experimental](https://img.shields.io/badge/-experimental-blue) |

| Attribute  | Type | Description  | Examples  | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) | Stability |
|---|---|---|---|---|---|
| [`db.collection.name`](/docs/attributes-registry/db.md) | string | Cosmos DB container name. [1] | `public.users`; `customers` | `Conditionally Required` If available | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
| [`db.cosmosdb.consistency_level`](/docs/attributes-registry/db.md) | string | Account or request [consistency level](https://learn.microsoft.com/azure/cosmos-db/consistency-levels). | `Eventual`; `ConsistentPrefix`; `BoundedStaleness`; `Strong`; `Session` | `Conditionally Required` If available. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |

> | [`db.cosmosdb.regions_contacted`](/docs/attributes-registry/db.md) | string | List of regions contacted by the client during a given operation. | `North Central US,Central US`; `East US 2,North Central US` | `Conditionally Required` If available | ![Experimental](https://img.shields.io/badge/-experimental-blue) |

Reviewer: Is there a possibility that this will be too high cardinality for a metric? Since this is a string, I'm assuming `North Central US, Central US` would be different than `Central US, North Central US`. Is this guaranteed to be in order of which regions were actually contacted first second etc.?
